### PR TITLE
CORE-1791: Remove Spring security plugin

### PR DESCRIPTION
**Pull request by kare**

### CODE REVIEW

File: grails-app/conf/Config.groovy [hunk 1]
```diff
@@ -187,7 +187,9 @@ environments {
 /**
  * API & CORS config
  */
-cors.url.pattern = ['/api/*', '/contact/send', '/profile/*', '/logout/*', '/login/*', '/j_spring_security_check', '/canvas', '/logout', '/j_spring_security_logout']
```
Comment (kare) [inline grails-app/conf/Config.groovy:190]: Is this ok? I don't think that anything else is needed here.

Comment (kare) [reply grails-app/conf/Config.groovy:190]: @hpihkala Are views grails-app/views/canvas/embed.gsp and grails-app/views/canvas/editor.gsp currently in use?

Comment (hpihkala) [reply grails-app/conf/Config.groovy:190]: @kare No they are not used, in fact they shouldn't even exist. All `gsp`s except the email templates should have been deleted. Sometimes they come back accidentally in merges, that might be the case here too.

File: grails-app/controllers/com/unifina/controller/api/AuthApiController.groovy [hunk 2]
```diff
@@ -171,9 +166,8 @@ class AuthApiController {
 			return render([success: false, error: userService.beautifyErrors(command.errors.getAllErrors())] as JSON)
 		}
 
-		String salt = saltSource instanceof NullSaltSource ? null : registrationCode.username
 		RegistrationCode.withTransaction { status ->
-			user.password = springSecurityService.encodePassword(command.password, salt)
+			user.password = passwordEncoder.encodePassword(command.password, salt)
```
Comment (hpihkala) [inline grails-app/controllers/com/unifina/controller/api/AuthApiController.groovy:170]: @kare **major finding, looks broken:**
- `salt` is not defined
- `encodePassword(String, String)` is not defined. Correct method signature is `encodePassword(String password)`.

File: grails-app/services/com/unifina/service/UserService.groovy [hunk 3]
```diff
@@ -96,27 +91,14 @@ class UserService {
...
-			roles = roleClass.findAllByAuthorityInList(secConf.ui.register.defaultRoleNames)
-			if (roles.size() != secConf.ui.register.defaultRoleNames.size()) {
-				throw new RuntimeException("Roles not found: " + secConf.ui.register.defaultRoleNames)
-			}
-		}
-
-		roles.each { role ->
-			userRoleClass.create user, role
+	def addRoles(SecUser user, List<SecRole> roles = null) {
+		roles.each { SecRole role ->
```
Comment (hpihkala) [inline grails-app/services/com/unifina/service/UserService.groovy:95]: **Major finding:**
NPE if `roles == null` (which it is by default). Please do `roles?.each`

File: grails-app/services/com/unifina/service/UserService.groovy [hunk 4]
```diff
@@ -189,15 +171,14 @@ class UserService {
...
 
 	SecUser getUserFromUsernameAndPassword(String username, String password) throws InvalidUsernameAndPasswordException {
-		PasswordEncoder encoder = new BCryptPasswordEncoder()
 		SecUser user = SecUser.findByUsername(username)
 		if (user == null) {
 			throw new InvalidUsernameAndPasswordException("Invalid username or password")
 		}
-		String dbHash = user.password
-		if (encoder.matches(password, dbHash)) {
+		String dbHash = user.password // TODO: $2a$10$z0HZdlGT7tvG6TSw4r/3Z.kqxJO4yM/ON4zX1pQ4TR1Kj3aidO/6q
```
Comment (hpihkala) [inline grails-app/services/com/unifina/service/UserService.groovy:178]: What's this TODO?

File: grails-app/conf/BuildConfig.groovy [hunk 5]
```diff
@@ -83,6 +83,23 @@ grails.project.dependency.resolution = {
 			excludes('org.springframework:spring-context:*')
 			excludes('org.springframework:spring-orm:*')
 		}
+		compile('org.springframework.security:spring-security-core:3.2.9.RELEASE') {
```
[the remark this answers is not in the document]
Comment (kare) [reply grails-app/conf/BuildConfig.groovy:86]: Bcrypt implementation comes from spring-security-core.

File: grails-app/services/com/unifina/service/UserService.groovy (hunk 4 above)
Comment (kare) [reply grails-app/services/com/unifina/service/UserService.groovy:178]: I was testing password encryption and forgot about the comment.

File: grails-app/conf/BuildConfig.groovy (hunk 5 above)
[the remark this answers is not in the document]
Comment (hpihkala) [reply grails-app/conf/BuildConfig.groovy:86]: Please add comment

**Pull request merged**